NSS-Day-Cohort-69/tabloid-hamburgers
# Ft/edit post

Adds the ability to edit posts you've made.

-added endpoint for editing post
-added DTO for editing post
-created edit post link and page
-moved post list navlink to be outside is admin ternary
-added UserIsAuthor component, which checks if the user is the author of the post and returns its children if they are
-fixed issue where invalid data was passed with creating post for the image url
-fixed issue where create post and edit post weren't sending you to post list

To test:

1. pull and checkout branch, restart API
2. sign in as any user (experience is the same for admins and authors)
3. ensure edit button only appears for posts you've created
4. ensure you it does not allow you to edit posts that weren't written by user
5. ensure you are able to access edit form for posts the user has written
6. ensure all fields are populated with existing data
7. make some changes to these fields, click submit, return to post list to ensure relevant data has changed

closes #39 

merge #59 before this
